### PR TITLE
[BD-30] Refactor and optimize GroupServer endpoints

### PR DESCRIPTION
**Pull request by stephenjusto247**
* Refactored group server endpoints by TRYING to adhere to REST conventions (https://restfulapi.net/resource-naming/)
* Created groupServerDao, textChannelDao, chatLogDao, and inviteDao
* Modified GroupServer, TextChannel, ChatLog, and Invite models
* Fixed frontend bugs encountered as a result of the above backend changes

PS it might be best for you guys to clear/drop your guys' local mongodb since the existing groupservers and other data will probably no longer work correctly

### CODE REVIEW

File: client/src/components/groupserver/GroupServer.jsx [hunk 1]
```diff
@@ -19,21 +19,20 @@ export default function GroupServer() {
...
     if (groupServerId) {
       try {
-        await fetch('/api/groupServer/verify', {
-          method: 'POST',
+        await fetch(`/api/group-servers/${groupServerId}`, {
+          method: 'GET',
           headers: {
             'Content-Type': 'application/json',
           },
+          /*
```
Comment (NRDuya) [inline client/src/components/groupserver/GroupServer.jsx:27]: Is this body/comment needed?

Comment (NRDuya) [inline client/src/components/groupserver/GroupServer.jsx:27]: Is this body/comment necessary?

File: server/routes/groupServerRoutes.js [hunk 2]
```diff
@@ -1,439 +1,327 @@
...
+  const timeLapse = (currentDate.getTime() - creationDate.getTime()) / 60000;
+  return invite.expiration <= 0 || (timeLapse < invite.expiration && invite.limit !== 0);
+}
+
+async function decreaseInviteUse(invite) {
+  if (invite.limit) {
+    await AddNumberToInviteUse(invite._id, -1);
+    if (invite.limit - 1 <= 0) await deleteInvite({ _id: invite._id });
   }
 }
```
Comment (NRDuya) [inline server/routes/groupServerRoutes.js:71]: Leave in separate utils. File should be full of routes only

File: client/src/components/groupserver/GroupServer.jsx [hunk 3]
```diff
@@ -19,21 +19,20 @@ export default function GroupServer() {
...
           body: JSON.stringify({
-            type: 'verify',
-            userId: user._id,
             groupServerId,
             textChannelId: textChannelId || Object.keys(state.groupServers[groupServerId].textChannels)[0],
           }),
```
Comment (NRDuya) [inline client/src/components/groupserver/GroupServer.jsx:31]: Necessary comment?

**Review by NRDuya — CHANGES_REQUESTED** [review]
Slight formatting issues, but everything else LGTM🥇👍

File: client/src/components/groupserver/GroupServer.jsx (hunk 3 above)
Comment (stephenjusto247) [reply client/src/components/groupserver/GroupServer.jsx]: whoops

File: server/routes/groupServerRoutes.js (hunk 2 above)
Comment (stephenjusto247) [reply server/routes/groupServerRoutes.js]: done

**Review by NRDuya — APPROVED** [review]
💯

**Pull request merged**